src/lib/dnssd/Resolver.h

```diff
@@ -35,67 +35,115 @@
-            mAddress[i].ToString(addrBuffer);
-            ChipLogProgress(Discovery, "    Addr %u: [%s]:%u", i, addrBuffer, mPort);
-        }
-#endif // CHIP_PROGRESS_LOGGING
-    }
+    size_t numIPs = 0; // number of valid IP addresses
+    Inet::IPAddress ipAddress[kMaxIPAddresses];
+
+    uint16_t port                         = 0;
+    char hostName[kHostNameMaxLength + 1] = {};
+    bool supportsTcp                      = false;
+    Optional<System::Clock::Milliseconds32> mrpRetryIntervalIdle;
+    Optional<System::Clock::Milliseconds32> mrpRetryIntervalActive;
+
+    CommonResolutionData() { Reset(); }
+
+    bool IsValid() const { return !IsHost("") && (numIPs > 0) && (ipAddress[0] != chip::Inet::IPAddress::Any); }
 
     ReliableMessageProtocolConfig GetMRPConfig() const
     {
         const ReliableMessageProtocolConfig defaultConfig = GetLocalMRPConfig();
         return ReliableMessageProtocolConfig(GetMrpRetryIntervalIdle().ValueOr(defaultConfig.mIdleRetransTimeout),
                                              GetMrpRetryIntervalActive().ValueOr(defaultConfig.mActiveRetransTimeout));
     }
-    Optional<System::Clock::Milliseconds32> GetMrpRetryIntervalIdle() const { return mMrpRetryIntervalIdle; }
-    Optional<System::Clock::Milliseconds32> GetMrpRetryIntervalActive() const { return mMrpRetryIntervalActive; }
+    Optional<System::Clock::Milliseconds32> GetMrpRetryIntervalIdle() const { return mrpRetryIntervalIdle; }
+    Optional<System::Clock::Milliseconds32> GetMrpRetryIntervalActive() const { return mrpRetryIntervalActive; }
 
     bool IsDeviceTreatedAsSleepy(const ReliableMessageProtocolConfig * defaultMRPConfig) const
     {
         // If either sleepy interval (Idle - SII, Active - SAI) has a value and that value is greater
         // than the value passed to this function, then the peer device will be treated as if it is
         // a Sleepy End Device (SED)
-        return (mMrpRetryIntervalIdle.HasValue() && (mMrpRetryIntervalIdle.Value() > defaultMRPConfig->mIdleRetransTimeout)) ||
-            (mMrpRetryIntervalActive.HasValue() && (mMrpRetryIntervalActive.Value() > defaultMRPConfig->mActiveRetransTimeout));
+        return (mrpRetryIntervalIdle.HasValue() && (mrpRetryIntervalIdle.Value() > defaultMRPConfig->mIdleRetransTimeout)) ||
+            (mrpRetryIntervalActive.HasValue() && (mrpRetryIntervalActive.Value() > defaultMRPConfig->mActiveRetransTimeout));
+    }
+
+    bool IsHost(const char * host) const { return strcmp(host, hostName) == 0; }
+
+    void Reset()
+    {
+        memset(hostName, 0, sizeof(hostName));
+        mrpRetryIntervalIdle   = NullOptional;
+        mrpRetryIntervalActive = NullOptional;
+        numIPs                 = 0;
+        port                   = 0;
+        supportsTcp            = false;
+        interfaceId            = Inet::InterfaceId::Null();
+        for (unsigned i = 0; i < kMaxIPAddresses; ++i)
+        {
+            ipAddress[i] = chip::Inet::IPAddress::Any;
```

Comment:
```suggestion
        for (auto & addr : ipAddress)
        {
            addr = chip::Inet::IPAddress::Any;
```

Reply:
Updated, however this was existing code. Somewhat afraid to not only move things around but also add bugs.